src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -35,11 +36,35 @@ public InstanceController(IEnvironment environment, IInstanceManager instanceMan
         [HttpPost]
         [Route("admin/instance/assign")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContext encryptedAssignmentContext)
+        public async Task<IActionResult> Assign([FromBody] HostAssignmentRequest hostAssignmentContext)
         {
-            _logger.LogDebug($"Starting container assignment for host : {Request?.Host}. ContextLength is: {encryptedAssignmentContext.EncryptedContext?.Length}");
+            if (string.IsNullOrEmpty(hostAssignmentContext.EncryptedContext) &&
+                hostAssignmentContext.AssignmentContext is null)
+            {
+                return BadRequest("At least one of 'assignmentContext' or 'encryptedContext' must be provided.");
+            }
+
+            if (!string.IsNullOrEmpty(hostAssignmentContext.EncryptedContext) &&
+                hostAssignmentContext.AssignmentContext is not null)
+            {
+                return BadRequest("Only one of 'assignmentContext' or 'encryptedContext' may be set.");
+            }
+
+            if (!string.IsNullOrEmpty(hostAssignmentContext.EncryptedContext))
+            {
+                _logger.LogDebug("Starting container assignment. ContextLength is {ContextLength}", hostAssignmentContext.EncryptedContext.Length);
+            }
+            else
+            {
+                if (!User.HasClaim(c => c.Type == SecurityConstants.AssignUnencryptedClaimType && c.Value == "true"))
+                {
+                    _logger.LogWarning("Required claims missing for invoking unencrypted assignment");
+                    return StatusCode(StatusCodes.Status403Forbidden);
```

Comment:
Tiny nit. There are a bunch of helper methods from `ControllerBase`. Or is there a reason you don't want to use `Forbid()`?

```suggestion
                    return Forbid();
```

Reply:
Done